### PR TITLE
Step 1 - Tripal Importer class

### PR DESCRIPTION
**Pull request by risharde**
This is an absolute basic PR that just contains the Tripal Importer class. More PRs to come.
@laceysanderson hopefully this is an okay start to get code in?

### CODE REVIEW

**Comment by risharde** [comment]
Tagging @laceysanderson

File: tripal/includes/TripalImporter.inc [hunk 1]
```diff
@@ -0,0 +1,775 @@
...
+   *
+   * @param $import_id
+   *   The ID of the import recrod.
+   *
+   * @return
+   *   An TripalImporter object of the appropriate child class.
+   */
+  static public function byID($import_id) {
+    // Get the importer.
+    $import = db_select('tripal_import', 'ti')
```
Comment (laceysanderson) [inline tripal/includes/TripalImporter.inc]: We no longer support Drupal 8.9.x. db_select and all other db_* procedural calls have been deprecated for 9.0.x and should be replaced. The change record for this with instructions to update can be found here: https://www.drupal.org/node/2993033

File: tripal/includes/TripalImporter.inc [hunk 2]
```diff
@@ -0,0 +1,775 @@
...
+   * Loads an existing import record into this object.
+   *
+   * @param $import_id
+   *   The ID of the import record.
+   */
+  public function load($import_id) {
+    $class = get_called_class();
+
+    // Get the importer.
+    $import = db_select('tripal_import', 'ti')
```
Comment (laceysanderson) [inline tripal/includes/TripalImporter.inc]: See deprecation comment above. Change as mentioned in https://www.drupal.org/node/2993033

File: tripal/includes/TripalImporter.inc [hunk 3]
```diff
@@ -0,0 +1,775 @@
...
+    // If we have a job then pass along the messaging to the job.
+    if ($this->job) {
+      // $this->job->logMessage($message, $variables, $severity);
+      $this->job->log($message);
+    }
+    // If we don't have a job then just use the drpual_set_message.
+    else {
+      // Report this message to watchdog or set a message.
+      if ($severity == TRIPAL_CRITICAL or $severity == TRIPAL_ERROR) {
+        drupal_set_message($tmessage, 'error');
```
Comment (laceysanderson) [inline tripal/includes/TripalImporter.inc]: Drupal_set_message is deprecated in 9.0.x. See https://www.drupal.org/node/2774931 for how to change.

File: tripal/includes/TripalImporter.inc [hunk 4]
```diff
@@ -0,0 +1,775 @@
...
+      }
+      if ($severity == TRIPAL_WARNING) {
+        drupal_set_message($tmessage, 'warning');
```
Comment (laceysanderson) [inline tripal/includes/TripalImporter.inc]: Drupal_set_message is deprecated in 9.0.x. See https://www.drupal.org/node/2774931 for how to change.

**Review by laceysanderson — CHANGES_REQUESTED** [review]
A few deprecated functions mentioned in the review comments are to be changed but otherwise, a quick code review looks good. Once the comments are fixed, this will be ready for merge.

**Comment by risharde** [comment]
Changes have been made though I wish there was a way to actually test this code @laceysanderson

**Review by laceysanderson — APPROVED** [review]
Changes according to code review look good 👍 

I changed the base of this PR to an intermediate code reviewed importer code branch.

Here's how I see this workflow going (also sent in slack):
1. For any outstanding functionality, Rish creates a new branch based on the `tv4-tripalImporterReviewed` branch, makes commits on that new branch and then creates a PR
to merge the new branch back into `tv4-tripalImporterReviewed`
2. Lacey will do a quick code review on each importer PR and once the code looks good will merge the PR
3. Once all importer functionality has made it into small PRs that were then code-reviewed and merged into `tv4-tripalImporterReviewed`, Rish can test the `tv4-tripalImporterReviewed` branch for working functionality and tests.
4. Each time Rish finds problems in the `tv4-tripalImporterReviewed` branch during testing, he will create a new small problem-focused PR similar to those in step 1.
5. Finally, once Rish feels the `tv4-tripalImporterReviewed` branch is bug-free and all problem-focused PRs are merged, Rish can create a PR merging `tv4-tripalImporterReviewed` into `9.x-4.x`.

**Pull request merged**